openedx/core/djangoapps/user_api/learner_skill_levels/views.py

```diff
@@ -0,0 +1,109 @@
+
+from .api import get_learner_skill_levels
+from .utils import get_top_skill_categories_for_job, get_job_holder_usernames
+
+
+class LearnerSkillLevelsView(APIView):
+    """
+        **Use Cases**
+
+            Returns top 5 job categories for the given job. Checks which skill the user has learned via courses
+            and assign scores to each skill in category. Also takes first 100 users in our system to calculate
+            average score for each category.
+
+        **Request format**
+
+            GET /api/user/v1/skill_level/{job_id}/
+
+        **Response Values for GET**
+
+            If the specified job_id doesn't exist, an HTTP
+            404 "Not Found" response is returned.
+
+            If a logged in user makes a request with an existing job, an HTTP 200
+            "OK" response is returned that contains a JSON string.
+
+        **Example Request**
+
+            GET /api/user/v1/skill_level/1/
+
+        **Example Response**
+
+            {
+                "job": "Digital Product Manager",
+                "skill_categories": [
+                    {
+                        "name": "Information Technology",
+                        "id": 1,
+                        "user_score": 0.4, // request user's score
+                        "edx_average_score": 0.7,
+                        "skills_subcategories": [
+                            {
+                                "id": 1,
+                                "name": "Databases",
+                                "skills": [
+                                    {"id": 2, "name": "Query Languages", "score": 1},
+                                    {"id": 3, "name": "MongoDB", "score": 3},
+                                ]
+                            },
+                            {
+                                "id": 2,
+                                "name": "IT Management",
+                                "skills": [
+                                    {"id": 1, "name": "Technology Roadmap", "score": 2},
+                                ]
+                            },
+                            // here remaining job related skills subcategories
+                        ]
+                    },
+
+                    // Here more 4 skill categories
+                ]
+            }
+    """
+    authentication_classes = (
+        JwtAuthentication,
+        BearerAuthenticationAllowInactiveUser,
+        SessionAuthenticationAllowInactiveUser
+    )
+    permission_classes = (permissions.IsAuthenticated, )
+
+    def get(self, request, job_id):
+        """
+        GET /api/user/v1/skill_level/{job_id}/
+        """
+        # get top categories for the given job
+        response = get_top_skill_categories_for_job(job_id)
+
+        job_holder_usernames = get_job_holder_usernames(job_id)
+        users = User.objects.filter(username__in=job_holder_usernames['usernames'])
```

Comment:
Were is the logic to make sure if users are less than threshold 5, we should not be calculating edx_average_score?

Reply:
Good catch, will add it.